metadata_backend/api/operators.py

```diff
@@ -707,32 +707,83 @@ def __init__(self, db_client: AsyncIOMotorClient) -> None:
+    async def get_submission_field(self, submission_id: str, field: str) -> Union[str, list, dict]:
+        """Get the only a field from the submission.
 
         :param submission_id: internal accession ID of submission
-        :returns: project ID submission is associated to
+        :param field: field name
+        :returns: field value
         """
         try:
-            submission_cursor = self.db_service.query("submission", {"submissionId": submission_id})
+            submission_cursor = self.db_service.query(
+                "submission", {"submissionId": submission_id}, {"_id": False, field: 1}, limit=1
+            )
             submissions = [submission async for submission in submission_cursor]
         except (ConnectionFailure, OperationFailure) as error:
             reason = f"Error happened while getting submission, err: {error}"
             LOG.exception(reason)
-            raise web.HTTPBadRequest(reason=reason)
+            raise web.HTTPInternalServerError(reason=reason) from error
+        except AttributeError as error:
+            reason = f"Submission '{submission_id}' doesn't have the requested '{field}' field."
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason) from error
 
         if len(submissions) == 1:
             try:
-                return submissions[0]["projectId"]
+                return submissions[0][field]
             except KeyError as error:
                 # This should not be possible and should never happen, if the submission was created properly
-                reason = f"Submission: '{submission_id}' does not have an associated project, err: {error}"
+                reason = f"Submission: '{submission_id}' does not have a value for '{field}', err: {error}"
                 LOG.exception(reason)
-                raise web.HTTPBadRequest(reason=reason)
-        else:
-            reason = f"Submission: '{submission_id}' not found."
-            LOG.error(reason)
-            raise web.HTTPBadRequest(reason=reason)
+                raise web.HTTPBadRequest(reason=reason) from error
+
+        reason = f"Submission: '{submission_id}' not found."
+        LOG.error(reason)
+        raise web.HTTPBadRequest(reason=reason)
+
+    async def get_submission_field_str(self, submission_id: str, field: str) -> str:
+        """Get the only a string field from the submission.
+
+        :param submission_id: internal accession ID of submission
+        :param field: field name
+        :returns: string value
+        """
+        value = await self.get_submission_field(submission_id, field)
+        if isinstance(value, str):
+            return value
+
+        reason = (
+            f"Submission: '{submission_id}' has an invalid {field}, "
+            f"expected 'str', got {type(value)}. This is a bug."
+        )
+        LOG.error(reason)
+        raise web.HTTPInternalServerError(reason=reason)
```

Comment:
Based on what happens in `get_submission_field()`, I'm not sure this edge case can even be reached but might as well have it just in case.

Reply:
That error should be raised if you'd try to get a different type of field with that method. So the method expected a string, but you asked for a boolean. The exception will show there was a mistake.